src/build.d

```diff
@@ -256,49 +297,52 @@ auto defineBackend()
 }
 
 /// Returns: the dependencies that generate required string files: VERSION and SYSCONFDIR.imp
-alias stringFiles = memoize!defineStringFiles;
-auto defineStringFiles()
+alias versionFile = memoize!defineVersionFile;
+auto defineVersionFile()
 {
     const versionFile = env["G"].buildPath("VERSION");
     auto commandFunction = (){
         "(TX) VERSION".writeln;
         ["git", "describe", "--dirty"].runCanThrow.toFile(versionFile);
     };
-    Dependency versionDependency = {
+    Dependency dependency = {
         target: versionFile,
         commandFunction: commandFunction,
     };
+    return new DependencyRef(dependency);
+}
+alias sysconfDirFile = memoize!defineSysconfDirFile;
+auto defineSysconfDirFile()
+{
     const sysconfDirFile = env["G"].buildPath("SYSCONFDIR.imp");
-    commandFunction = (){
+    auto commandFunction = (){
         "(TX) SYSCONFDIR".writeln;
         env["SYSCONFDIR"].toFile(sysconfDirFile);
     };
-    Dependency sysconfDirDependency = {
+    Dependency dependency = {
         sources: [thisBuildScript],
         target: sysconfDirFile,
         commandFunction: commandFunction,
     };
-    return [new DependencyRef(versionDependency), new DependencyRef(sysconfDirDependency)];
+    return new DependencyRef(dependency);
 }
 
+
 /**
-Main build routine for the DMD compiler.
-Defines the required order for the build dependencies, runs all these dependency dependencies
-and afterwards builds the DMD compiler.
+Defines the main rule to build the DMD compiler.
 
 Params:
   extra_flags = Flags to apply to the main build but not the dependencies
 */
-auto dmd(string[] extraFlags...) { return memoize!defineDmd(extraFlags); }
-auto defineDmd(string[] extraFlags...)
```

Comment:
Split defineDmd into `defineDmdExe` and `defineDmd`, where `defineDmd` will build the exe and the conf file.

Also made a `defineDmdUnittest` that doesn't need the conf file, and also builds dmd with a different exe name so it doesn't overwrite the non-unittest dmd exe.